OpenRA.Mods.Common/Traits/World/ShroudRenderer.cs

```diff
@@ -159,6 +175,65 @@ public ShroudRenderer(World world, ShroudRendererInfo info)
 			notVisibleEdges = info.UseExtendedIndex ? Edges.AllSides : Edges.AllCorners;
 		}
 
+		Sheet CopySpritesOntoSingleSheet()
+		{
+			var allSprites = shroudSprites.Concat(fogSprites);
+			var width = (int)Math.Ceiling(allSprites.Sum(sprite => sprite.Size.X));
+			var height = (int)Math.Ceiling(allSprites.Max(sprite => sprite.Size.Y));
+			using (var builder = new SheetBuilder(SheetType.BGRA, Exts.NextPowerOf2(new Size(width, height))))
```

Comment:
Yup. Addressed by taking a stab at the required size and retrying with progressively bigger sheets until everything gets rendered onto a single sheet.